### PR TITLE
Rate limit async change

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: As a follow up to [29199](https://github.com/envoyproxy/envoy/pull/29199) we are moving the rate limit filter to use the change.
Additional Description: The original API would hash a proto on every single request. This will pre hash the proto and store it within the filter callback.
Risk Level: Low

### CODE REVIEW

**Comment by DiazAlan** [comment]
@ggreenway Another follow up to the AsyncClientManager change. Moving over rate limiting

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc [hunk 1]
```diff
@@ -123,13 +123,13 @@ void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::strin
 }
 
 ClientPtr rateLimitClient(Server::Configuration::FactoryContext& context,
-                          const envoy::config::core::v3::GrpcService& grpc_service,
+                          const Grpc::GrpcServiceConfigWithHashKey config_with_hash_key,
```
Comment (pradeepcrao) [inline source/extensions/filters/common/ratelimit/ratelimit_impl.cc:126]: s/Grpc::GrpcServiceConfigWithHashKey/Grpc::GrpcServiceConfigWithHashKey&

File: source/extensions/filters/common/ratelimit/ratelimit_impl.h [hunk 2]
```diff
@@ -81,7 +81,7 @@ class GrpcClientImpl : public Client,
  * Builds the rate limit client.
  */
 ClientPtr rateLimitClient(Server::Configuration::FactoryContext& context,
-                          const envoy::config::core::v3::GrpcService& grpc_service,
+                          const Grpc::GrpcServiceConfigWithHashKey config_with_hash_key,
```
Comment (pradeepcrao) [inline source/extensions/filters/common/ratelimit/ratelimit_impl.h:84]: s/Grpc::GrpcServiceConfigWithHashKey/Grpc::GrpcServiceConfigWithHashKey&

File: source/extensions/filters/http/rate_limit_quota/client_impl.h [hunk 3]
```diff
@@ -32,12 +32,13 @@ class RateLimitClientImpl : public RateLimitClient,
                                 envoy::service::rate_limit_quota::v3::RateLimitQuotaResponse>,
                             public Logger::Loggable<Logger::Id::rate_limit_quota> {
 public:
-  RateLimitClientImpl(const envoy::config::core::v3::GrpcService& grpc_service,
+  RateLimitClientImpl(Grpc::GrpcServiceConfigWithHashKey config_with_hash_key,
```
Comment (pradeepcrao) [inline source/extensions/filters/http/rate_limit_quota/client_impl.h:35]: s/Grpc::GrpcServiceConfigWithHashKey/Grpc::GrpcServiceConfigWithHashKey&

**Review by pradeepcrao — COMMENTED** [review]
Thank you for working on this @DiazAlan !

File: source/extensions/filters/http/rate_limit_quota/client_impl.h (hunk 3 above)
Comment (DiazAlan) [reply source/extensions/filters/http/rate_limit_quota/client_impl.h:35]: Done

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc (hunk 1 above)
Comment (DiazAlan) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc]: Done

File: source/extensions/filters/http/rate_limit_quota/client_impl.h (hunk 3 above)
Comment (ggreenway) [inline source/extensions/filters/http/rate_limit_quota/client_impl.h:35]:
```suggestion
  RateLimitClientImpl(const Grpc::GrpcServiceConfigWithHashKey& config_with_hash_key,
```

**Review by ggreenway — CHANGES_REQUESTED** [review]
Looks good, one small nit

/wait

File: source/extensions/filters/http/rate_limit_quota/client_impl.h (hunk 3 above)
Comment (DiazAlan) [reply source/extensions/filters/http/rate_limit_quota/client_impl.h]: Done

**Review by ggreenway — APPROVED** [review]

**Pull request merged**